src/components/Table/Table.qa.stories.tsx

```diff
@@ -0,0 +1,332 @@
+import { HasIdAndName } from "src/types";
+import { noop } from "src/utils";
+import { zeroTo } from "src/utils/sb";
+
+export default {
+  component: GridTable,
+  title: "Design QA/Table",
+  parameters: { layout: "fullscreen", backgrounds: { default: "white" } },
+} as Meta;
+
+type BeamData = {
+  id: string;
+  favorite: boolean;
+  status: string;
+  commitmentName: string;
+  tradeCategories: string[];
+  location: string;
+  date: string;
+  priceInCents: number;
+};
+type BeamRow = SimpleHeaderAndDataWith<BeamData>;
+
+export function Fixed() {
+  return (
+    <GridTable<BeamRow>
+      style={beamFixedStyle}
+      sorting={{ on: "client", initial: [1, "ASC"] }}
+      columns={beamStyleColumns()}
+      rows={[
+        { kind: "header", id: "header" },
+        ...zeroTo(20).map((idx) => ({
+          kind: "data" as const,
+          id: `r:${idx + 1}`,
+          data: {
+            id: `r:${idx + 1}`,
+            favorite: idx < 5,
+            commitmentName: idx === 9 ? "A longer name that will truncate with an ellipsis" : `Commitment ${idx + 1}`,
+            date: `01/${idx + 1 > 9 ? idx + 1 : `0${idx + 1}`}/2020`,
+            status: "Success",
+            tradeCategories: ["Roofing", "Architecture", "Plumbing"],
+            priceInCents: 1234_56 + idx,
+            location: idx % 2 ? "l:1" : "l:2",
+          },
+        })),
+      ]}
+    />
+  );
+}
+
+export function Flexible() {
+  return (
+    <GridTable<BeamRow>
+      style={beamFlexibleStyle}
+      sorting={{ on: "client", initial: [1, "ASC"] }}
+      columns={beamStyleColumns()}
+      rows={[
+        { kind: "header", id: "header" },
+        ...zeroTo(20).map((idx) => ({
+          kind: "data" as const,
+          id: `r:${idx + 1}`,
+          data: {
+            id: `r:${idx + 1}`,
+            favorite: idx < 5,
+            commitmentName: idx === 9 ? "A longer name that will truncate with an ellipsis" : `Commitment ${idx + 1}`,
+            date: `01/${idx + 1 > 9 ? idx + 1 : `0${idx + 1}`}/2020`,
+            status: "Success",
+            tradeCategories: ["Roofing", "Architecture", "Plumbing"],
+            priceInCents: 1234_56 + idx,
+            location: idx % 2 ? "l:1" : "l:2",
+          },
+        })),
+      ]}
+    />
+  );
+}
+
+type BeamBudgetData = {
+  name?: ReactNode;
+  original?: number;
+  changeOrders?: number;
+  reallocations?: number;
+  revised?: number;
+  committed?: number;
+  difference?: number;
+  actuals?: number;
+  projected?: number;
+  costToComplete?: number;
+  children?: BeamChildRow[];
+};
+type HeaderRow = { kind: "header" };
+type BeamTotalsRow = { kind: "totals"; id: string } & BeamBudgetData;
+type BeamParentRow = { kind: "parent"; id: string } & BeamBudgetData;
+type BeamChildRow = { kind: "child"; id: string } & BeamBudgetData;
+type BeamNestedRow = BeamTotalsRow | HeaderRow | BeamParentRow | BeamChildRow;
+
+export function Nested() {
```

Comment:
Maybe go ahead and have flexible nested and fixed nested stories? Afaiu nested could be either fixed or flexible?